x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -120,7 +120,9 @@
                             // To facilitate ML UI functionality being controlled using Kibana security privileges
                             "manage_ml",
                             // The symbolic constant for this one is in SecurityActionMapper, so not accessible from X-Pack core
-                            "cluster:admin/analyze"
+                            "cluster:admin/analyze",
+                            // To facilitate using the file uploader functionality
+                            "monitor_text_structure"
```

Comment:
I am not 100% about this. It seems that we might need this for the Kibana security privileges?

Reply:
I think this is correct.

@peteharverson is having a meeting with the maps team about moving the UI functionality out of the ML app. How the permissions will be done will be a key decision to make. But I think accessing this endpoint as the Kibana system user after checking the most appropriate Kibana permission will be most intuitive.

Reply:
The ML UI currently makes all calls to `_ml/find_file_structure` as the Kibana system user, so adding in `monitor_text_structure` for the Kibana system user is correct here.

Reply:
awesome! FWIW, I verified that this works through some testing locally.